mne/time_frequency/tfr.py

```diff
@@ -281,13 +292,17 @@ def cwt_morlet(X, sfreq, freqs, use_fft=True, n_cycles=7.0, zero_mean=False,
     """
     mode = 'same'
     # mode = "valid"
-    n_signals, n_times = X.shape
+    if not all(isinstance(x, int) or x is None for x in time_range):
+        raise TypeError("time_range elements should be integers or None.")
+    tmin, tmax = time_range
+    n_signals, n_times = X[:, tmin:tmax].shape
```

Comment:
Here is only defined the size of the time dimension.
The lines where the "cropping" [tmin:tmax] is done in practice is in _cwt() (lines 245 and 249) and is applied after tfr estimation, avoiding edge artifacts.